File: android/src/main/java/to/holepunch/bare/android/ErrorCodes.kt
package to.holepunch.bare.android

/**
 * Canonical error codes, mirrored from `core/messages/constants.ts`.
 *
 * Keep this file in sync with the TypeScript `ErrorCode` union; the parity
 * test [ErrorCodesParityTest] enforces the contract.
 */
object ErrorCodes {
    const val UNSUPPORTED_CAPABILITY = "UNSUPPORTED_CAPABILITY"
    const val UNSUPPORTED_EVENT = "UNSUPPORTED_EVENT"
    const val HOST_ERROR = "HOST_ERROR"
    const val TRANSPORT_ERROR = "TRANSPORT_ERROR"
    const val PLUGIN_ERROR = "PLUGIN_ERROR"
    const val INVALID_RESPONSE = "INVALID_RESPONSE"
    const val FRAME_TOO_LARGE = "FRAME_TOO_LARGE"
    const val FRAME_INVALID = "FRAME_INVALID"
    const val TIMEOUT = "TIMEOUT"
}

File: ios/Sources/BareHost/ErrorCodes.swift
import Foundation

/**
 * Canonical error codes, mirrored from `core/messages/constants.ts`.
 *
 * Keep this file in sync with the TypeScript `ErrorCode` union and
 * `android/.../ErrorCodes.kt`; the parity test enforces the contract.
 */
public enum ErrorCodes {
  public static let unsupportedCapability = "UNSUPPORTED_CAPABILITY"
  public static let unsupportedEvent = "UNSUPPORTED_EVENT"
  public static let hostError = "HOST_ERROR"
  public static let transportError = "TRANSPORT_ERROR"
  public static let pluginError = "PLUGIN_ERROR"
  public static let invalidResponse = "INVALID_RESPONSE"
  public static let frameTooLarge = "FRAME_TOO_LARGE"
  public static let frameInvalid = "FRAME_INVALID"
  public static let timeout = "TIMEOUT"
}
